arian1917/arian1917/Program.cs

```diff
@@ -0,0 +1,579 @@
+    Type 1 - to Track Drinking water.
+    Type 2 - to Track Running counting.
+    Type 3 - to Track Programming learning hours.
+    Type 4 - To Create a new Custom tracker.
+    Type 5 - To Track the Custom tracker.
+            ");
+            
+            string? readResult = Console.ReadLine(); 
+
+            switch(readResult){
+                case "0":
+                    Environment.Exit(0); 
+                    break; 
+                case "1":
+                    GetUserInput(new DrinkingWater()); 
+                    break; 
+                case "2": 
+                    GetUserInput(new RunningCount());
+                    break;
+                case "3":
+                    GetUserInput(new ProgrammingHours());
+                    break; 
+                case "4":
+                    CreateCustomHabit(); 
+                    break; 
+                case "5":
+                    TrackCustomHabit(); 
+                    break; 
+                default:
+                    Console.WriteLine("Invalid key, please try again"); 
+                    HabitSelection(); 
+                    break; 
+            }
+        }
+        
+        static void CreateCustomHabit(){
+            Console.Clear(); 
+            
+            Console.WriteLine("Which habit you want to track?"); 
+            string? habitName = Console.ReadLine();
+
+            while(String.IsNullOrEmpty(habitName)){
+                Console.WriteLine("Please enter a valid input"); 
+                Console.ReadLine(); 
+            }
+
+            Console.WriteLine("Which unit you want to use?"); 
+            string? measureUnit = Console.ReadLine(); 
+            while(String.IsNullOrEmpty(measureUnit)){
+                Console.WriteLine("Please enter a valid input"); 
+                Console.ReadLine(); 
+            }
+
+            SaveCustomHabit(habitName, measureUnit);
+
+            Console.WriteLine($"Custom habit '{habitName}' with measure unit '{measureUnit}' has been created.");
+            Console.WriteLine("Press any key to return to the main menu.");
+            Console.ReadKey();
+            HabitSelection(); 
+        }
+
+        static void SaveCustomHabit(string habitName, string measureUnit){
+            string tableName = $"habit_{habitName.ToLower().Replace(" ", "_")}";
+
+            using (var connection = new SqliteConnection(connectionString)){
+                connection.Open();
+                var tableCmd = connection.CreateCommand();
+
+                tableCmd.CommandText =
+                    @$"CREATE TABLE IF NOT EXISTS custom_habits (
+                        Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                        Name TEXT, 
+                        MeasureUnit TEXT, 
+                        TableName TEXT
+                    )";
+
+                tableCmd.ExecuteNonQuery();
+
+                tableCmd.CommandText = 
+                    @$"CREATE TABLE IF NOT EXISTS {tableName} (
```

Comment:
Note: This is open for [SQL Injection attacks](https://www.stackhawk.com/blog/net-sql-injection-guide-examples-and-prevention/)

Try to use parameterised queries.